netket/experimental/driver/tdvp.py

```diff
@@ -0,0 +1,395 @@
+# Copyright 2020, 2021  The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+from typing import Callable, Optional, Sequence, Union
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+from tqdm import tqdm
+
+import netket as nk
+from netket.driver import AbstractVariationalDriver
+from netket.driver.abstract_variational_driver import _to_iterable
+from netket.driver.vmc_common import info
+from netket.experimental.dynamics.runge_kutta import euclidean_norm, maximum_norm
+from netket.logging.json_log import JsonLog
+from netket.operator import AbstractOperator
+from netket.optimizer import LinearOperator
+from netket.optimizer.qgt import QGTAuto
+from netket.utils import mpi
+from netket.utils.dispatch import dispatch
+from netket.utils.types import PyTree
+from netket.vqs import VariationalState, VariationalMixedState, MCState
+
+from netket.experimental.dynamics import RKIntegratorConfig
+
+
+@dispatch
+def dwdt(state, driver, t, w, *, stage=None):
```

Comment:
this in theory is part of the interface (probably not, but still...) so maybe we should find a better name?

Reply:
Let's keep `dwdt` for the time being. We can change it later, if we come up with a better name.